gpxpy/parser.py

```diff
@@ -165,7 +165,7 @@ def __init__(self, xml_or_file=None, parser=None):
 
     def init(self, xml_or_file):
         text = xml_or_file.read() if hasattr(xml_or_file, 'read') else xml_or_file
-        if text[:3] == "\xEF\xBB\xBF": #Remove utf-8 Byte Order Mark (BOM) if present
+        if text.startswith(u'\xEF\xBB\xBF'):  # Remove utf-8 Byte Order Mark (BOM) if present
```

Comment:
Yes, from the tests I guess now. 

But from xml_or_file.read() it's really not obvious if we are working with bytes or unicode here in Py 2 and 3.